azurelinuxagent/common/logcollector.py

```diff
@@ -0,0 +1,360 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+from datetime import datetime
+import glob
+from heapq import heappush, heappop
+import os
+import tarfile
+import zipfile
+
+from azurelinuxagent.common.future import ustr
+from azurelinuxagent.common.utils.fileutil import mkdir, read_file, rm_files, append_file, append_items_to_file
+from azurelinuxagent.common.utils.shellutil import run_command
+from azurelinuxagent.common.utils.textutil import safe_shlex_split
+
+
+LOG_COLLECTOR_DIR = '/var/lib/waagent/logcollector'
+TRUNCATED_FILES_DIR = '/var/truncated'
+
+OUTPUT_RESULTS_FILE_PATH = os.path.join(LOG_COLLECTOR_DIR, "results.txt")
+OUTPUT_ARCHIVE_PATH = os.path.join(LOG_COLLECTOR_DIR, "logs.tar")
+COMPRESSED_ARCHIVE_PATH = os.path.join(LOG_COLLECTOR_DIR, "logs.zip")
+
+MUST_COLLECT_FILES = [
+    '/var/log/waagent*',
+    '/var/lib/waagent/GoalState.*.xml',
+    '/var/lib/waagent/waagent_status.json',
```

Comment:
we should avoid hardcoding /var/lib/waagent, etc. maybe the idea of matching the Inspect Disk manifest does not work as well as we expected. ID uses static paths, but our log collector should be more dynamic, allowing us to change those directories should we wish to (customer have been asking us to get out of /var, for example)

Reply:
I agree with the sentiment and will add support for that, but I would like to keep the hard-coded values for the v1 of this feature for two reasons:
1) `/var/lib/waagent` is still the most commonly used location for agent files (I haven't seen this modified once so far)
2) it would be beneficial to release the v1 for this feature as soon as possible so we can get real feedback and telemetry from live VMs and users quicker and then decide on areas for improvement.

Reply:
what's the logic for postponing it?

Reply:
Added since we already have the API for this